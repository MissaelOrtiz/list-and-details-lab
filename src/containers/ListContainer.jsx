import React, { Component } from 'react';
import { fetchApi } from '../components/services/fetchApi';
import ListComponent from '../components/ListAndDetails/ListComponent';
import LoadingComponent from '../components/ListAndDetails/LoadingComponent';

export default class ListContainer extends Component {
    state = {
      characters: [],
      loading: true,
    }

    componentDidMount() {
      fetchApi().then(characters => { 
        this.setState({ characters, loading: false });
      });
    }

    render() {
      if(this.state.loading) {
        return (<LoadingComponent />);
      }
      return <ListComponent characters={this.state.characters} />;
    }
}
