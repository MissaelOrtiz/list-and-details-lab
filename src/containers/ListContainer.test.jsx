import React from 'react';
import { render, screen } from '@testing-library/react';
import ListContainer from './ListContainer';

describe('ListContainer', () => {
  it('renders a list of characters to the page', async () => {
    await render(<ListContainer />);
    screen.getByAltText('loading spinner');
    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).toMatchSnapshot();
  });
});
