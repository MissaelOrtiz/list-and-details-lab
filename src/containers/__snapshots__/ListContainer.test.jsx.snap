// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer renders a list of characters to the page 1`] = `
<ul
  aria-label="characters"
  role="list"
  style="display: flex; flex-direction: row; width: 100%; flex-wrap: wrap; text-align: center; justify-content: center;"
>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Admiral"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
      />
      <p>
        Admiral
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Agent S"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
      />
      <p>
        Agent S
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Agnes"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
      />
      <p>
        Agnes
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Al"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
      />
      <p>
        Al
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Alfonso"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
      />
      <p>
        Alfonso
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Alice"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
      />
      <p>
        Alice
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Ankha"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
      />
      <p>
        Ankha
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Alli"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
      />
      <p>
        Alli
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Anicotti"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
      />
      <p>
        Anicotti
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Angus"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
      />
      <p>
        Angus
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Amelia"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
      />
      <p>
        Amelia
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Annalisa"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
      />
      <p>
        Annalisa
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Antonio"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
      />
      <p>
        Antonio
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Annalise"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
      />
      <p>
        Annalise
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Anabelle"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
      />
      <p>
        Anabelle
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Apollo"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
      />
      <p>
        Apollo
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Anchovy"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
      />
      <p>
        Anchovy
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Apple"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
      />
      <p>
        Apple
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Astrid"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
      />
      <p>
        Astrid
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Audie"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
      />
      <p>
        Audie
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Aurora"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
      />
      <p>
        Aurora
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Ava"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
      />
      <p>
        Ava
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Avery"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
      />
      <p>
        Avery
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Baabara"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/df/Baabara_NH.png/revision/latest?cb=20200802165552"
      />
      <p>
        Baabara
      </p>
    </button>
  </li>
  <li
    style="list-style-type: none; display: flex; flex-direction: column; justify-content: center; align-items: center; text-align: center; margin: 20px;"
  >
    <button
      style="width: 300px; height: 50vh; font-weight: bolder; font-size: 2rem; cursor: pointer;"
    >
      <img
        alt="Barold"
        height="200px"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
      />
      <p>
        Barold
      </p>
    </button>
  </li>
</ul>
`;
